core/src/main/resources/hudson/model/Label/Badge/index.jelly

```diff
@@ -0,0 +1,30 @@
+<!--
+The MIT License
+-->
+
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:l="/lib/layout" >
+  <st:adjunct includes="io.jenkins.plugins.bootstrap5"/>
+  
+  <j:set var="cssSuffix" value="${it.name.replace(':', '-').replace('.', '-')}"/>
+  <j:set var="modelLink" value="${modelLink}"/>
+  <j:if test="${modelLink == null}">
+    <j:set var="modelLink" value="model-link"/>
+  </j:if>
+
+  <style>
+  .badge-mine-${cssSuffix} {
+    <j:if test="${it.color != null &amp;&amp; !it.color.isEmpty()}">
+      background: ${it.color};
```

Comment:
Looks like potential CSS injection from people able to configure label colors (Overall/Manage?).

Reply:
Attackers can patch the page to change the UI element shown, or edit the HTTP request sent to Jenkins.

Reply:
Interesting ... How shall / can be solved it?

Reply:
Server-side validation: Ignore invalid submissions (don't throw exceptions, Jenkins is weird with them), use a known good (default) value instead if you get one.